src/NClient/NClient.Abstractions/Providers/Transport/Common/Content.cs

```diff
@@ -22,7 +22,7 @@ public class Content : IContent
         public Content(Stream? streamContent = null, string? encoding = null, IMetadataContainer? headerContainer = null)
         {
             Stream = streamContent ?? new MemoryStream(Array.Empty<byte>());
-            Encoding = string.IsNullOrEmpty(encoding) ? Encoding.UTF8 : Encoding.GetEncoding(encoding);
+            Encoding = string.IsNullOrEmpty(encoding) ? null : Encoding.GetEncoding(encoding);
```

Comment:
Are you sure this changes won't cause NRE-like exception when request contains json-data and files and etc?

Reply:
Now we can use `Microsoft.AspNetCore.Http.IFormFile` interface to transport files. This interface doesn't have encoding, so I made `Encoding` field optional

Reply:
And again.. It doesn't answered to my question. That about possible of NRE?

Reply:
I think you should formulate the questions more clearly. What exactly are you afraid of? That current code throw an NRE exception? Answer is no - we have tests. The future code may throw an NRE exception? Again, we have tests, in addition, we have a warnings to check null values